### PR TITLE
[INT-1139] Add 'innerAuthorizationUrl' to oauth template

### PR DESCRIPTION
**Pull request by ewelinaBS**
**Ticket:**
https://dronedeploy.atlassian.net/browse/INT-1139

**Work done:**
New field is added into oauth config template. Implementation is described in `READ.me` file. Http request is called with `node-fetch`.

### CODE REVIEW

File: handlers/authorizationCode.js [hunk 1]
```diff
@@ -26,13 +28,24 @@ const refreshHandler = (req, res, ctx) => {
       const storageTokenInfo = getStorageTokenInfo(req, ctx);
 
       return accessTokensTable.getRowByExternalId(storageTokenInfo.externalId)
-        .then((result) => {
+        .then(async (result) => {
```
Comment (frank-ais) [inline handlers/authorizationCode.js]: I not sure but I believe that in this project we were not using async/await for backward compatibility.

File: handlers/authorizationCode.js [hunk 2]
```diff
@@ -26,13 +28,24 @@ const refreshHandler = (req, res, ctx) => {
...
           if (!result.ok) {
             if (couldNotFindData(result)) {
               return res.status(401).send(packageError(result));
             }
             return res.status(500).send(packageError(result));
           }
+
+          if (!!result.data.accessToken) {
```
Comment (frank-ais) [inline handlers/authorizationCode.js]: Wouldn't it be more straightforward to have check for `provider.innerAuthorizationUrl` here, before calling `getInnerAuthorizationRespone`?

File: handlers/authorizationCode.js [hunk 3]
```diff
@@ -70,6 +83,33 @@ const refreshHandler = (req, res, ctx) => {
     })
 };
 
+const getInnerAuthorizationResponse = async (accessToken) => {
+    let authorized = true;
+    const url = provider.innerAuthorizationUrl;
+    if (url) {
+        let options = {
+            method: 'GET',
+            withCredentials: true,
```
Comment (frank-ais) [inline handlers/authorizationCode.js]: Are you sure that we need `withCredentials` and `credentials` options? It seems that the Authorization header should be sufficient.

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@dronedeploy/oauth-function-template",
-  "version": "1.3.0",
+  "version": "1.3.1",
```
Comment (frank-ais) [inline package.json]: Adding `innerAuthorizationUrl` looks like a new feature, so according to symver we should bump minor version number (1.3.0 -> 1.4.0).

File: handlers/authorizationCode.js (hunk 1 above)
Comment (ewelinaBS) [reply handlers/authorizationCode.js]: @frank-ais What do you think about my current solution? As you suggested, I removed `async/await` and replaced it with `Promise.then`

**Pull request merged**